solr/core/src/java/org/apache/solr/servlet/CoreContainerProvider.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.solr.servlet;
+
+import org.apache.solr.core.CoreContainer;
+
+public interface CoreContainerProvider {
```

Comment:
Wouldn't Suppler<CoreContainer> suffice?

Reply:
this interface has been eliminated in the most recent version Not sure how you're reviewing old versions here. In fact this has now become the name of the class instead. Pls see https://github.com/apache/solr/pull/397/files#diff-0bfd58ac1dc3b66fdda09a3fedf2c1b1fe37f57d9a981201e440bdc05108864eR1